connect/src/routes/automaticTokenBridgeRoute.ts

```diff
@@ -0,0 +1,103 @@
+import { Network } from "@wormhole-foundation/sdk-base";
+import { Signer, TokenTransferDetails, isSameToken } from "@wormhole-foundation/sdk-definitions";
+import { TokenTransfer } from "../protocols/tokenTransfer";
+import { TransferReceipt, TransferState } from "../wormholeTransfer";
+import { AutomaticRoute, ValidationError, ValidationResult } from "./route";
+
+export namespace AutomaticTokenBridgeRoute {
+  export type Options = {
+    nativeGasDropoff?: bigint;
+  };
+}
+
+export class AutomaticTokenBridgeRoute<N extends Network> extends AutomaticRoute<
+  N,
+  AutomaticTokenBridgeRoute.Options
+> {
+  NATIVE_GAS_DROPOFF_SUPPORTED = true;
+
+  async isSupported(): Promise<boolean> {
+    // No transfers to same chain
+    if (this.request.from.chain === this.request.to.chain) return false;
+
+    // No transfers to unsupported chains
+    const fromChain = this.wh.getChain(this.request.from.chain);
+    if (!fromChain.supportsAutomaticTokenBridge()) return false;
+
+    const toChain = this.wh.getChain(this.request.to.chain);
+    if (!toChain.supportsAutomaticTokenBridge()) return false;
+
+    const atb = await fromChain.getAutomaticTokenBridge();
+    if (!atb.isRegisteredToken(this.request.source.address)) return false;
+
+    // If the destination token was set, and its different than what
+    // we'd get from a token bridge transfer, then this route is not supported
+    if (this.request.destination) {
+      const destToken = await TokenTransfer.lookupDestinationToken(
+        fromChain,
+        toChain,
+        this.request.source,
+      );
+      if (!isSameToken(destToken, this.request.destination)) return false;
+    }
+
+    return true;
+  }
+
+  async isAvailable(): Promise<boolean> {
+    // TODO
+    return true;
+  }
+
+  getDefaultOptions(): AutomaticTokenBridgeRoute.Options {
+    return { nativeGasDropoff: 0n };
+  }
+
+  async validate(
+    options: AutomaticTokenBridgeRoute.Options,
+  ): Promise<ValidationResult<boolean, ValidationError>> {
+    const transfer = this.toTransferDetails(options);
+    try {
+      await TokenTransfer.validateTransferDetails(this.wh, transfer);
+      return { valid: true };
+    } catch (e) {
+      return { valid: false, error: "Some other error" };
```

Comment:
maybe `ValidationResult` should allow `error` to be `any`?

Reply:
I just made it `Error` since we have it already there